itest/rust/src/engine_tests/gfile_test.rs

```diff
@@ -0,0 +1,137 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::io::{BufRead, Read, Seek, SeekFrom, Write};
+
+use crate::framework::itest;
+use godot::builtin::GString;
+use godot::engine::{file_access::ModeFlags, GFile};
+
+const TEST_FULL_PATH: &str = "res://file_tests";
+
+fn remove_test_file() {
+    let test_file_path = std::path::Path::new(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/../godot/",
+        "file_tests"
+    ));
+    std::fs::remove_file(test_file_path)
+        .unwrap_or_else(|_| panic!("Couldn't remove test file: {}", test_file_path.display()));
+}
+
+#[itest]
+fn non_unique_error_works() {
+    use godot::engine::NotUniqueError;
+    use godot::engine::RefCounted;
+
+    let unique = RefCounted::new();
+    assert!(NotUniqueError::check(unique).is_ok());
+
+    let shared = RefCounted::new();
+    let cloned = shared.clone();
+    assert!(NotUniqueError::check(shared).is_err());
+    if let Err(error) = NotUniqueError::check(cloned) {
+        assert_eq!(error.get_reference_count(), 2);
+    }
```

Comment:
Avoid checking the same condition twice.
```suggestion
match NotUniqueError::check(cloned) {
    Err(error) => assert_eq!(error.get_reference_count(), 2),
    Ok(_) => panic!("NotUniqueError::check must not succeed"),
}
```